### ISSUE TITLE
[Microsoft AVD] Authentication result: SEC_E_INVALID_TOKEN

### ISSUE THREAD

**Issue by Brueggus** [FreeRDP/FreeRDP#9363]
**Describe the bug**
Connecting to my Microsoft Azure Virtual Desktop VM fails with the following error (full log below):

```
[16:22:06:441] [57043:0000ded4] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x55d833ff91d0]: CONNECTION_STATE_NEGO --> CONNECTION_STATE_AAD
[16:22:06:448] [57043:0000ded4] [TRACE][com.freerdp.core.rdp] - [rdp_recv_callback][0x55d833ff91d0]: (client)[CONNECTION_STATE_AAD -> CONNECTION_STATE_AAD] current return STATE_RUN_SUCCESS [0] [feeding 158 bytes, 0 bytes not processed]
[16:22:06:473] [57043:0000ded4] [ERROR][com.freerdp.aad] - [aad_parse_state_auth]: Authentication result: SEC_E_INVALID_TOKEN (0x80090308)
[16:22:06:473] [57043:0000ded4] [ERROR][com.freerdp.core.rdp] - [rdp_recv_callback_int][0x55d833ff91d0]: CONNECTION_STATE_AAD - aad_recv() fail
[16:22:06:473] [57043:0000ded4] [ERROR][com.freerdp.core.rdp] - [rdp_recv_callback_int][0x55d833ff91d0]: CONNECTION_STATE_AAD status STATE_RUN_FAILED [-1]
```

**To Reproduce**
1. Go to https://client.wvd.microsoft.com/arm/webclient/index.html
2. Download .rdpw file
3. Launch `sdl-freerdp Downloads/My\ Personal\ AVD.rdpw /gateway:type:arm /u:my.user@domain.com /f`
4. Authenticate
5. Authenticate again 
6. sdl-freerdp quits: `Authentication result: SEC_E_INVALID_TOKEN (0x80090308)`

**Expected behavior**
The connection to the AVD should be successful.

**Screenshots**
n/a

**Log**

[freerdp.log](https://github.com/FreeRDP/FreeRDP/files/12514632/freerdp.log)

(I tried to redact any sensitive values. If important information is missing, please let me know.)

**Application details**
* FreeRDP version (`This is FreeRDP version 3.0.0-dev4 (5f8f69512)`)
* Command line used `sdl-freerdp Downloads/My\ Personal\ AVD.rdpw /gateway:type:arm /u:user@domain.com /f /log-level:TRACE`
* Output of `xfreerdp /buildconfig`
```
[16:37:55:731] [58028:0000e2ac] [WARN][com.freerdp.client.common.cmdline] - [freerdp_client_warn_experimental]: [experimental] sdl-freerdp client is currently experimental!
[16:37:55:731] [58028:0000e2ac] [WARN][com.freerdp.client.common.cmdline] - [freerdp_client_warn_experimental]:  If problems occur please check https://github.com/FreeRDP/FreeRDP/issues for know issues or create a new one!
[16:37:55:731] [58028:0000e2ac] [WARN][com.freerdp.client.common.cmdline] - [freerdp_client_warn_experimental]:  Developers hang out in https://matrix.to/#/#FreeRDP:matrix.org?via=matrix.org - dont hesitate to ask some questions. (replies might take some time depending on your timezone)
This is FreeRDP version 3.0.0-dev4 (5f8f69512)
Build configuration: BUILD_TESTING=OFF WINPR_HAVE_AIO_H=1 WINPR_HAVE_EXECINFO_BACKTRACE=1 WINPR_HAVE_EXECINFO_BACKTRACE_SYMBOLS=1 WINPR_HAVE_EXECINFO_BACKTRACE_SYMBOLS_FD=1 WINPR_HAVE_EXECINFO_HEADER=1 WINPR_HAVE_FCNTL_H=1 WINPR_HAVE_GETLOGIN_R=1 WINPR_HAVE_GETPWUID_R=1 WINPR_HAVE_INTTYPES_H=1 WINPR_HAVE_JOURNALD_H=TRUE WINPR_HAVE_POLL_H=1 WINPR_HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIB=1 WINPR_HAVE_PTHREAD_MUTEX_TIMEDLOCK_LIBS= WINPR_HAVE_PTHREAD_MUTEX_TIMEDLOCK_SYMBOL=1 WINPR_HAVE_STDBOOL_H=1 WINPR_HAVE_STDINT_H=1 WINPR_HAVE_STRNDUP=1 WINPR_HAVE_SYSLOG_H=1 WINPR_HAVE_SYS_EVENTFD_H=1 WINPR_HAVE_SYS_FILIO_H= WINPR_HAVE_SYS_SELECT_H=1 WINPR_HAVE_SYS_SOCKIO_H= WINPR_HAVE_SYS_TIMERFD_H=1 WINPR_HAVE_TM_GMTOFF=1 WINPR_HAVE_UNISTD_H=1 WINPR_HAVE_UNWIND_H=1 WITH_AAD=ON WITH_ADD_PLUGIN_TO_RPATH=OFF WITH_ALSA=ON WITH_CAIRO=OFF WITH_CCACHE=ON WITH_CHANNELS=ON WITH_CLANG_FORMAT=ON WITH_CLIENT=ON WITH_CLIENT_AVAILABLE=1 WITH_CLIENT_CHANNELS=ON WITH_CLIENT_CHANNELS_AVAILABLE=1 WITH_CLIENT_COMMON=ON WITH_CLIENT_INTERFACE=OFF WITH_CLIENT_SDL=ON WITH_CLIENT_SDL_AVAILABLE=1 WITH_CUPS=ON WITH_DEBUG_ALL=OFF WITH_DEBUG_CAPABILITIES=OFF WITH_DEBUG_CERTIFICATE=OFF WITH_DEBUG_CHANNELS=OFF WITH_DEBUG_CLIPRDR=OFF WITH_DEBUG_CODECS=OFF WITH_DEBUG_DVC=OFF WITH_DEBUG_EVENTS=OFF WITH_DEBUG_KBD=OFF WITH_DEBUG_LICENSE=OFF WITH_DEBUG_MUTEX=OFF WITH_DEBUG_NEGO=OFF WITH_DEBUG_NLA=OFF WITH_DEBUG_NTLM=OFF WITH_DEBUG_RAIL=OFF WITH_DEBUG_RDP=OFF WITH_DEBUG_RDPDR=OFF WITH_DEBUG_RDPEI=OFF WITH_DEBUG_RDPGFX=OFF WITH_DEBUG_REDIR=OFF WITH_DEBUG_RFX=OFF WITH_DEBUG_RINGBUFFER=OFF WITH_DEBUG_SCARD=OFF WITH_DEBUG_SCHANNEL=OFF WITH_DEBUG_SDL_EVENTS=OFF WITH_DEBUG_SDL_KBD_EVENTS=OFF WITH_DEBUG_SND=OFF WITH_DEBUG_SVC=OFF WITH_DEBUG_SYMBOLS=OFF WITH_DEBUG_THREADS=OFF WITH_DEBUG_TIMEZONE=OFF WITH_DEBUG_TRANSPORT=OFF WITH_DEBUG_TSG=OFF WITH_DEBUG_TSMF=OFF WITH_DEBUG_TSMF_AVAILABLE=0 WITH_DEBUG_URBDRC=OFF WITH_DEBUG_WND=OFF WITH_DEBUG_X11=OFF WITH_DEBUG_X11_LOCAL_MOVESIZE=OFF WITH_DEBUG_XV=OFF WITH_DSP_EXPERIMENTAL=OFF WITH_DSP_FFMPEG=ON WITH_DSP_FFMPEG_AVAILABLE=1 WITH_EVENTFD_READ_WRITE=1 WITH_FAAC=OFF WITH_FAAD2=OFF WITH_FFMPEG=ON WITH_FREERDP_DEPRECATED=OFF WITH_FREERDP_DEPRECATED_COMMANDLINE=OFF WITH_FUSE=ON WITH_GFX_H264=ON WITH_GPROF=OFF WITH_GSM=OFF WITH_ICU=ON WITH_INTERNAL_MD4=OFF WITH_INTERNAL_MD5=OFF WITH_INTERNAL_RC4=OFF WITH_IPP=OFF WITH_JPEG=ON WITH_KRB5=ON WITH_KRB5_NO_NTLM_FALLBACK=OFF WITH_LAME=OFF WITH_LIBRARY_VERSIONING=ON WITH_LIBSYSTEMD=ON WITH_LODEPNG=OFF WITH_MACAUDIO=OFF WITH_MACAUDIO_AVAILABLE=0 WITH_MANPAGES=ON WITH_MBEDTLS=OFF WITH_NATIVE_SSPI=OFF WITH_OPENCL=OFF WITH_OPENH264=OFF WITH_OPENSC_PKCS11_LINKED=OFF WITH_OPENSLES=OFF WITH_OPENSSL=ON WITH_OSS=ON WITH_PCSC=ON WITH_PKCS11=ON WITH_PLATFORM_SERVER=ON WITH_POLL=ON WITH_PROFILER=OFF WITH_PROXY=ON WITH_PROXY_APP=ON WITH_PROXY_EMULATE_SMARTCARD=OFF WITH_PROXY_MODULES=ON WITH_PULSE=ON WITH_RDTK=ON WITH_SAMPLE=ON WITH_SANITIZE_ADDRESS=OFF WITH_SANITIZE_ADDRESS_AVAILABLE=1 WITH_SANITIZE_MEMORY=OFF WITH_SANITIZE_MEMORY_AVAILABLE=1 WITH_SANITIZE_THREAD=OFF WITH_SANITIZE_THREAD_AVAILABLE=1 WITH_SERVER=ON WITH_SERVER_CHANNELS=ON WITH_SERVER_INTERFACE=ON WITH_SHADOW=ON WITH_SMARTCARD_EMULATE=ON WITH_SMARTCARD_INSPECT=OFF WITH_SMARTCARD_PCSC=ON WITH_SOXR=OFF WITH_SSE2=ON WITH_SWSCALE=ON WITH_THIRD_PARTY=OFF WITH_UNICODE_BUILTIN=OFF WITH_VAAPI=OFF WITH_VAAPI_AVAILABLE=1 WITH_VALGRIND_MEMCHECK=OFF WITH_VALGRIND_MEMCHECK_AVAILABLE=1 WITH_VERBOSE_WINPR_ASSERT=ON WITH_VIDEO_FFMPEG=ON WITH_VIDEO_FFMPEG_AVAILABLE=1 WITH_WAYLAND=ON WITH_WEBVIEW=OFF WITH_WEBVIEW_QT=ON WITH_WINPR_DEPRECATED=OFF WITH_WINPR_TOOLS=ON WITH_WIN_CONSOLE=ON WITH_X11=ON WITH_XCURSOR=ON WITH_XEXT=ON WITH_XFIXES=ON WITH_XI=ON WITH_XINERAMA=ON WITH_XRANDR=ON WITH_XRENDER=ON WITH_XV=ON
Build type:          Release
CFLAGS:              -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions         -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security         -fstack-clash-protection -fcf-protection -fPIC -Wall -fvisibility=hidden -Wimplicit-function-declaration -Wredundant-decls -fno-omit-frame-pointer
Compiler:            GNU, 13.2.1
Target architecture: x64
```

* OS version connecting to (server side)
Windows 10
* If available the log output from a run with `/log-level:trace 2>&1 | tee log.txt` 
[freerdp.log](https://github.com/FreeRDP/FreeRDP/files/12514632/freerdp.log)

* If you built it yourself add some notes which tag/commit/branch you have used, also your cmake parameters and
  compiler can help
Built from current master: [5f8f695](https://github.com/FreeRDP/FreeRDP/commit/5f8f69512b1307b341ef6cca1043b12679568bc4)
Used build file from https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=freerdp-git and added `-DWITH_WEBVIEW=OFF`

**Environment (please complete the following information):**
 - OS: Arch Linux
 - Version/Distribution: n/a
 - Architecture: [amd64, arm]: amd64

**Comment by hardening** [comment]
Does it work better if you ship the password on the command line with `/p:<pass>` ?

**Comment by Brueggus** [comment]
No, doesn't make a difference.

**Comment by hardening** [comment]
@Brueggus do you see any difference with https://github.com/hardening/FreeRDP/tree/fixes_9363 (with or without password) ?

**Comment by Brueggus** [comment]
@hardening Thanks for your help!
I've rebuilt with your patch applied but still get the same error. It's the same whether or not I supply the password.

```
[12:12:31:161] [28343:00006eb8] [DEBUG][com.freerdp.utils.http] - [freerdp_http_request]: body:
grant_type=srv_challenge
[12:12:31:291] [28343:00006eb8] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x55f98747e1d0]: CONNECTION_STATE_NEGO --> CONNECTION_STATE_AAD
[12:12:31:293] [28343:00006eb8] [TRACE][com.freerdp.core.rdp] - [rdp_recv_callback][0x55f98747e1d0]: (client)[CONNECTION_STATE_AAD -> CONNECTION_STATE_AAD] current return STATE_RUN_SUCCESS [0] [feeding 158 bytes, 0 bytes not processed]
[12:12:31:325] [28343:00006eb8] [ERROR][com.freerdp.aad] - [aad_parse_state_auth]: Authentication result: SEC_E_INVALID_TOKEN (0x80090308)
[12:12:31:325] [28343:00006eb8] [ERROR][com.freerdp.core.rdp] - [rdp_recv_callback_int][0x55f98747e1d0]: CONNECTION_STATE_AAD - aad_recv() fail
[12:12:31:325] [28343:00006eb8] [ERROR][com.freerdp.core.rdp] - [rdp_recv_callback_int][0x55f98747e1d0]: CONNECTION_STATE_AAD status STATE_RUN_FAILED [-1]
[12:12:31:325] [28343:00006eb8] [TRACE][com.freerdp.core.rdp] - [rdp_recv_callback][0x55f98747e1d0]: (client)[CONNECTION_STATE_AAD -> CONNECTION_STATE_AAD] current return STATE_RUN_FAILED [-1] [feeding 38 bytes, 0 bytes not processed]
[12:12:31:325] [28343:00006eb8] [ERROR][com.freerdp.core.transport] - [transport_check_fds]: transport_check_fds: transport->ReceiveCallback() - STATE_RUN_FAILED [-1]
[12:12:31:325] [28343:00006eb8] [DEBUG][com.freerdp.core.rdp] - [rdp_check_fds][0x55f98747e1d0]: transport_check_fds() - -1
[12:12:31:325] [28343:00006eb8] [ERROR][com.freerdp.core] - [rdp_client_wait_for_activation]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[12:12:31:325] [28343:00006eb8] [DEBUG][com.freerdp.core.rdp] - [rdp_finalize_reset_flags][0x55f98747e1d0]: [CONNECTION_STATE_AAD] reset finalize_sc_pdus
[12:12:31:325] [28343:00006eb8] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x55f98747e1d0]: CONNECTION_STATE_AAD --> CONNECTION_STATE_INITIAL
[12:12:31:326] [28343:00006eb8] [TRACE][com.freerdp.api] - [freerdp_channels_process_message]: IFCALL(message->Free) == NULL

➜  ~ sdl-freerdp /version                                                                                         
[12:12:40:308] [28597:00006fb5] [WARN][com.freerdp.client.common.cmdline] - [freerdp_client_warn_experimental]: [experimental] sdl-freerdp client is currently experimental!
[12:12:40:308] [28597:00006fb5] [WARN][com.freerdp.client.common.cmdline] - [freerdp_client_warn_experimental]:  If problems occur please check https://github.com/FreeRDP/FreeRDP/issues for know issues or create a new one!
[12:12:40:308] [28597:00006fb5] [WARN][com.freerdp.client.common.cmdline] - [freerdp_client_warn_experimental]:  Developers hang out in https://matrix.to/#/#FreeRDP:matrix.org?via=matrix.org - dont hesitate to ask some questions. (replies might take some time depending on your timezone)
This is FreeRDP version 3.0.0-dev4 (f37ddfefd)

```

**Comment by hardening** [comment]
@Brueggus with my branch  if you provide the password on the command line, does go through the RDSTLS path ?

**Comment by Brueggus** [comment]
I _think_ so:

```
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_set_negotiation_enabled]: Enabling security layer negotiation: TRUE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_set_restricted_admin_mode_required]: Enabling restricted admin mode: FALSE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_enable_rdp]: Enabling RDP security: FALSE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_enable_tls]: Enabling TLS security: FALSE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_enable_nla]: Enabling NLA security: FALSE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_enable_ext]: Enabling NLA extended security: FALSE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_enable_rdstls]: Enabling RDSTLS security: TRUE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_enable_aad]: Enabling RDS AAD security: TRUE
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x55f98747e1d0]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_connect]: state: NEGO_STATE_AAD
[12:12:22:914] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_attempt_rdsaad]: Attempting RDS AAD Auth security
[12:12:22:953] [28343:00006eb8] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[12:12:22:953] [28343:00006eb8] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer 51.116.225.43
[12:12:22:976] [28343:00006eb8] [DEBUG][com.freerdp.core.gateway.wst] - [wst_tls_connect]: connecting to rdgateway-c103-gwc-r1.wvd.microsoft.com 443
[12:12:22:120] [28343:00006eb8] [DEBUG][com.freerdp.core.gateway.wst] - [wst_handle_ok_or_forbidden]: Got Affinity cookie c94c9448ebdbc420b7bb374f768398bfe23a3b66f87815b2ae069d19116daafe
[12:12:22:123] [28343:00006eb8] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[12:12:22:123] [28343:00006eb8] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer 51.116.225.43
[12:12:22:145] [28343:00006eb8] [DEBUG][com.freerdp.core.gateway.wst] - [wst_tls_connect]: connecting to rdgateway-c103-gwc-r1.wvd.microsoft.com 443
[12:12:23:249] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_send_negotiation_request]: Adding terminating CRLF to routing token
[12:12:23:249] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_send_negotiation_request]: RequestedProtocols: 16
[12:12:23:311] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_process_negotiation_response]: RDP_NEG_RSP::flags = { [0x3f] |EXTENDED_CLIENT_DATA_SUPPORTED|DYNVC_GFX_PROTOCOL_SUPPORTED|RDP_NEGRSP_RESERVED|RESTRICTED_ADMIN_MODE_SUPPORTED|REDIRECTED_AUTHENTICATION_MODE_SUPPORTED|UNKNOWN }
[12:12:23:311] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_recv]: selected_protocol: 16
[12:12:23:311] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_attempt_rdsaad]: state: NEGO_STATE_FINAL
[12:12:23:311] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_connect]: Negotiated AAD security
[12:12:23:311] [28343:00006eb8] [DEBUG][com.freerdp.core.nego] - [nego_try_connect]: nego_security_connect with PROTOCOL_RDSAAD
```

**Comment by hardening** [comment]
@Brueggus I've updated the branch, can you retry with a git force update and with a password ?

**Comment by Brueggus** [comment]
Looks different now, but still no cigar. It's the same whether or not I supply the password.

```
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_enable_rdp]: Enabling RDP security: FALSE
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_enable_tls]: Enabling TLS security: FALSE
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_enable_nla]: Enabling NLA security: FALSE
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_enable_ext]: Enabling NLA extended security: FALSE
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_enable_rdstls]: Enabling RDSTLS security: TRUE
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_enable_aad]: Enabling RDS AAD security: FALSE
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x55d93b3fc1d0]: CONNECTION_STATE_INITIAL --> CONNECTION_STATE_NEGO
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_connect]: state: NEGO_STATE_RDSTLS
[15:31:43:226] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_attempt_rdstls]: Attempting RDSTLS security
[15:31:43:269] [55985:0000dab2] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[15:31:43:269] [55985:0000dab2] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer 51.116.225.43
[15:31:43:290] [55985:0000dab2] [DEBUG][com.freerdp.core.gateway.wst] - [wst_tls_connect]: connecting to rdgateway-c103-gwc-r1.wvd.microsoft.com 443
[15:31:43:421] [55985:0000dab2] [DEBUG][com.freerdp.core.gateway.wst] - [wst_handle_ok_or_forbidden]: Got Affinity cookie f119449f46dcef9bd5860a934b5295fe0a7cdf19ef369ded63982f197791bcee
[15:31:43:424] [55985:0000dab2] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: resetting error state
[15:31:43:424] [55985:0000dab2] [DEBUG][com.freerdp.core] - [freerdp_tcp_default_connect]: connecting to peer 51.116.225.43
[15:31:43:444] [55985:0000dab2] [DEBUG][com.freerdp.core.gateway.wst] - [wst_tls_connect]: connecting to rdgateway-c103-gwc-r1.wvd.microsoft.com 443
[15:31:43:537] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_send_negotiation_request]: Adding terminating CRLF to routing token
[15:31:43:538] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_send_negotiation_request]: RequestedProtocols: 5
[15:31:43:563] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_process_negotiation_response]: RDP_NEG_RSP::flags = { [0x3f] |EXTENDED_CLIENT_DATA_SUPPORTED|DYNVC_GFX_PROTOCOL_SUPPORTED|RDP_NEGRSP_RESERVED|RESTRICTED_ADMIN_MODE_SUPPORTED|REDIRECTED_AUTHENTICATION_MODE_SUPPORTED|UNKNOWN }
[15:31:43:563] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_recv]: selected_protocol: 4
[15:31:43:563] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_attempt_rdstls]: state: NEGO_STATE_FINAL
[15:31:43:563] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_connect]: Negotiated RDSTLS security
[15:31:43:563] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_try_connect]: nego_security_connect with PROTOCOL_RDSTLS
[15:31:43:661] [55985:0000dab2] [WARN][com.freerdp.crypto] - [verify_cb]: Certificate verification failure 'self-signed certificate (18)' at stack position 0
[15:31:43:661] [55985:0000dab2] [WARN][com.freerdp.crypto] - [verify_cb]: CN = VM80003000-2.redacted.net
[15:31:43:663] [55985:0000dab2] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_INITIAL	--> RDSTLS_STATE_CAPABILITIES
[15:31:43:663] [55985:0000dab2] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_CAPABILITIES	--> RDSTLS_STATE_AUTH_REQ
[15:31:43:663] [55985:0000dab2] [DEBUG][com.freerdp.core.rdstls] - [rdstls_set_state]: -- RDSTLS_STATE_AUTH_REQ	--> RDSTLS_STATE_AUTH_RSP
[15:31:43:695] [55985:0000dab2] [ERROR][com.freerdp.core.rdstls] - [rdstls_process_authentication_response]: resultCode: ERRBASE_UNKNOWN [0x0000052E] ERRBASE_UNKNOWN
[15:31:43:695] [55985:0000dab2] [ERROR][com.freerdp.core.transport] - [transport_connect_rdstls]: RDSTLS authentication failed
[15:31:43:695] [55985:0000dab2] [ERROR][com.freerdp.core] - [transport_connect_rdstls]: ERRCONNECT_AUTHENTICATION_FAILED [0x00020009]
[15:31:43:695] [55985:0000dab2] [DEBUG][com.freerdp.core.nego] - [nego_connect]: Failed to connect with RDSTLS security
[15:31:43:695] [55985:0000dab2] [DEBUG][com.freerdp.core.rdp] - [rdp_finalize_reset_flags][0x55d93b3fc1d0]: [CONNECTION_STATE_NEGO] reset finalize_sc_pdus
[15:31:43:695] [55985:0000dab2] [DEBUG][com.freerdp.core.rdp] - [rdp_client_transition_to_state][0x55d93b3fc1d0]: CONNECTION_STATE_NEGO --> CONNECTION_STATE_INITIAL
[15:31:43:697] [55985:0000dab2] [TRACE][com.freerdp.api] - [freerdp_channels_process_message]: IFCALL(message->Free) == NULL

```

**Comment by Brueggus** [comment]
@hardening I have now managed to connect... Thanks for your work!!
I had to use my "legacy" Active Directory username instead of the Azure AD username in the command line, i. e.
` sdl-freerdp Personal\ AVD.rdpw /gateway:type:arm /u:'DOMAIN\user' /p:'password'`

Is there a way to do this without supplying the password on the command line? I'd prefer not having my credentials all over my history file ;)
